webapp-django/crashstats/crashstats/forms.py

```diff
@@ -235,6 +235,16 @@ def clean_v(self):
             )
         return versions
 
+    def clean_date_end(self):
+        value = self.cleaned_data['date_end']
+        if value:
+            now = datetime.datetime.utcnow()
+            if not isinstance(value, datetime.datetime):
```

Comment:
Ugh. This whole thing is a mess. I think I discovered some weird things about django's `form.DateField()` and all was well until I had to hack around the tests. I'll take another stab at it. 

By the way, did you know that...:
```
>>> import datetime
>>> now = datetime.datetime.now()
>>> isinstance(now, datetime.datetime)
True
>>> isinstance(now, datetime.date)
True
```
...but...
```
>>> now = datetime.datetime.now().date()
>>> isinstance(now, datetime.date)
True
>>> isinstance(now, datetime.datetime)
False
```